include/envoy/stats/stats.h

```diff
@@ -225,41 +225,58 @@ class ParentHistogram : public virtual Histogram {
 typedef std::shared_ptr<ParentHistogram> ParentHistogramSharedPtr;
 
 /**
- * A sink for stats. Each sink is responsible for writing stats to a backing store.
+ * Provides sinks with access to stats during periodic stat flushes.
  */
-class Sink {
+class FlushSource {
```

Comment:
Is this specific to flush or do we want it to be used to access stats from admin, json etc in future? If we change the reset logic to trigger via `StoreRoot`, I guess it could be used at many places where stats are required. If that is useful, I think it would make sense to rename this as `StatsSource/StatsDataProvider/StatsProvider` to sound more generic rather than flush.